src/charm.py

```diff
@@ -268,6 +284,10 @@ def restart(self):
         if not self._container.exists(MIMIR_CONFIG_FILE):
             logger.error("cannot restart mimir: config file doesn't exist (yet).")
 
+        if not self._mimir_roles:
+            logger.warning("cannot restart mimir: no roles have been configured.")
```

Comment:
if no roles have been configured, has mimir ever even started?
should we instead raise, because we can't be restarting mimir if it's not running?

Reply:
So yeah this function is a bit weird; I added this guard as it was crashing whenever you'd relate the coordinator to the worker without roles

Since the previous function doesn't return (and since the same would happen if you relate to coordinator and then remove the relation), we should just exit this function at that point

If you haven't configured roles, Mimir won't be restarting; but it's normal and expected, cause the workload won't even be running. We shouldn't raise cause this is normal: a warning is there just to remind the user they have to configure the roles; we could remove it altogether and just `return`.

Reply:
mm no it makes sense. Assuming it will put `blocked` and stay there, perhaps we can downgrade it to `debug` and be done with it.